src/app/application/components/data-source/DataLoader.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+  <DataSource
+    v-slot="{ refresh }"
+    :src="props.src"
+    @change="(data) => srcData = data"
+    @error="(e) => srcError = e"
+  >
+    <template
+      v-if="allData.length > 0 && allData.every(item => typeof item !== 'undefined')"
+    >
+      <!-- if the data is loaded and then we get an error -->
+      <!-- keep the old data visible but also show a disconnected slot -->
+      <template
+        v-if="allErrors.length > 0"
+      >
+        <slot
+          name="disconnected"
+          :data="props.src !== '' ? allData[0] : undefined"
+          :errors="props.src !== '' ? allErrors[0] : undefined"
+          :refresh="props.src !== '' ? refresh : () => {}"
+        >
+          <!-- KAlert -->
+        </slot>
+      </template>
+      <slot
+        name="default"
+        :data="props.src !== '' ? allData[0] : undefined"
+        :errors="props.src !== '' ? allErrors[0] : undefined"
+        :refresh="props.src !== '' ? refresh : () => {}"
+      />
+    </template>
+
+    <template
+      v-else-if="allErrors.length > 0"
+    >
+      <slot
+        name="error"
+        :data="props.src !== '' ? allData[0] : undefined"
+        :errors="props.src !== '' ? allErrors[0] : undefined"
+        :refresh="props.src !== '' ? refresh : () => {}"
+      >
+        <ErrorBlock
+          :error="allErrors[0]"
+        />
+      </slot>
+    </template>
+    <template v-else>
+      <slot
+        v-if="props.loader"
+        name="connecting"
+        :data="props.src !== '' ? allData[0] : undefined"
+        :errors="props.src !== '' ? allErrors[0] : undefined"
+        :refresh="props.src !== '' ? refresh : () => {}"
+      >
+        <LoadingBlock />
+      </slot>
+      <slot
+        v-else
+        name="default"
+        :data="props.src !== '' ? allData[0] : undefined"
+        :errors="props.src !== '' ? allErrors[0] : undefined"
+        :refresh="props.src !== '' ? refresh : () => {}"
+      />
+    </template>
+  </DataSource>
+</template>
+<script lang="ts" setup>
+import { computed, ref } from 'vue'
+
+import ErrorBlock from '@/app/common/ErrorBlock.vue'
+import LoadingBlock from '@/app/common/LoadingBlock.vue'
+
+const props = withDefaults(defineProps<{
+  data: any[]
+  errors: (Error | undefined)[]
```

Comment:
**Question**: What’s the thought behind accepting data and error as arrays? We seem to use only the first one of them.